src/services/tracing.ts

```diff
@@ -171,6 +171,78 @@ export const asyncRunInSpan = <F extends () => ReturnType<F>>(
   return ret
 }
 
+// eslint-disable-next-line  @typescript-eslint/no-explicit-any
+export const wrapToRunInSpan = <A extends Array<any>, R>({
+  fn,
+  fnName,
+  namespace = "app",
+  spanAttributes = {},
+}: {
+  fn: (...args: A) => R
+  fnName?: string
+  namespace?: string
+  spanAttributes?: SpanAttributes
+}) => {
+  return (...args: A): R => {
+    const name = fnName || fn.name
```

Comment:
could we not always use `fn.name`? is there some use case where we specifically need to change the function name?

Reply:
removed